src/sequence.lisp

```diff
@@ -196,31 +196,39 @@
 (defun copy-seq (sequence)
   (subseq sequence 0))
 
-
-;;; Reduce (based on SBCL's version)
-
-(defun reduce (function sequence &key key from-end (start 0) end (initial-value nil ivp))
+(defun %reduce (function sequence key index from-end step-fn compare-fn end value)
```

Comment:
I guess that this tail-recursion could be problematic  when compiled to Javascript. We will likely get call stack overflow.